overlord/state/state.go

```diff
@@ -72,35 +72,48 @@ func (data customData) set(key string, value interface{}) {
 // The state is persisted on every unlock operation via the StateBackend
 // it was initialized with.
 type State struct {
-	// locking
 	mu  sync.Mutex
 	muC int32
-	// storage
+
 	backend Backend
 	data    customData
 	changes map[string]*Change
 	tasks   map[string]*Task
+
+	modified bool
 }
 
 // New returns a new empty state.
 func New(backend Backend) *State {
 	return &State{
-		backend: backend,
-		data:    make(customData),
-		changes: make(map[string]*Change),
-		tasks:   make(map[string]*Task),
+		backend:  backend,
+		data:     make(customData),
+		changes:  make(map[string]*Change),
+		tasks:    make(map[string]*Task),
+		modified: true,
```

Comment:
Because that state was never serialized into the backend before. Not doing that would break current test expectations which seem reasonable, for example.